src/client/jupyter/languageserver/notebookConverter.ts

```diff
@@ -127,9 +129,18 @@ export class NotebookConverter implements Disposable {
         if (wrapper) {
             // Diagnostics are supposed to be per file and are updated each time
             // Make sure to clear out old ones first
+            const cellUris: string[] = [];
+            const oldCellUris = this.mapOfConcatDocumentsWithCellUris.get(uri.toString()) || [];
             wrapper.notebook.getCells().forEach((c: NotebookCell) => {
                 result.set(c.document.uri, []);
+                cellUris.push(c.document.uri.toString());
             });
+            // Possible some cells were deleted, we need to clear the diagnostics of those cells as well.
+            const currentCellUris = new Set(cellUris);
+            oldCellUris
+                .filter((cellUri) => !currentCellUris.has(cellUri))
+                .forEach((cellUri) => result.set(Uri.parse(cellUri), []));
```

Comment:
Without this, if we delete a cell, the diagnostics of deleted cells remains & doesn't go away.